packages/now-cli/src/commands/env/pull.ts

```diff
@@ -0,0 +1,99 @@
+import chalk from 'chalk';
+import { NowContext, ProjectEnvTarget } from '../../types';
+import { Output } from '../../util/output';
+import Client from '../../util/client';
+import stamp from '../../util/output/stamp';
+import getEnvVariables from '../../util/env/get-env-records';
+import getDecryptedSecret from '../../util/env/get-decrypted-secret';
+import { getLinkedProject } from '../../util/projects/link';
+import cmd from '../../util/output/cmd';
+import withSpinner from '../../util/with-spinner';
+import { join } from 'path';
+import { promises } from 'fs';
+import { emoji, prependEmoji } from '../../util/emoji';
+const { writeFile } = promises;
+
+type Options = {
+  '--debug': boolean;
+};
+
+export default async function pull(
+  ctx: NowContext,
+  opts: Options,
+  args: string[],
+  output: Output
+) {
+  const {
+    authConfig: { token },
+    config,
+  } = ctx;
+  const { currentTeam } = config;
+  const { apiUrl } = ctx;
+  const debug = opts['--debug'];
+  const client = new Client({ apiUrl, token, currentTeam, debug });
+  const link = await getLinkedProject(output, client);
+
+  if (link.status === 'error') {
+    return link.exitCode;
+  } else if (link.status === 'not_linked') {
+    output.print(
+      `${chalk.red(
+        'Error!'
+      )} Your codebase isn’t linked to a project on ZEIT Now. Run ${cmd(
+        'now'
+      )} to link it.\n`
+    );
+    return 1;
+  } else {
+    const { project } = link;
+    if (args.length > 1) {
+      output.error(
+        `Invalid number of arguments. Usage: ${chalk.cyan(
+          '`now env pull <file>`'
+        )}`
+      );
+      return 1;
+    }
+
+    const [filename = '.env'] = args;
+    output.print(
+      `Downloading Development environment variables for project ${chalk.bold(
+        project.name
+      )}\n`
+    );
+    const pullStamp = stamp();
+
+    const records = await withSpinner('Downloading', async () => {
+      const dev = ProjectEnvTarget.Development;
+      const envs = await getEnvVariables(output, client, project.id, dev);
+      for (const env of envs) {
+        env.value = await getDecryptedSecret(output, client, env.value);
+      }
+      return envs;
+    });
+
+    const quote = '"';
+    const contents =
+      records
+        .map(
+          ({ key, value }) =>
+            `${key}=${quote}${escapeValue(value, quote)}${quote}`
+        )
+        .join('\n') + '\n';
+
+    const fullPath = join(process.cwd(), filename);
+    await writeFile(fullPath, contents, 'utf8');
```

Comment:
Should we append if a file already exists?

Reply:
I don't think so because you want the local file to reflect the cloud. And if we appended, then you couldn't run `now env pull` twice. I would expect to be able to `git pull && now env pull` and everything be ready for `now dev`.

Reply:
My concern is the case where we overwrite the user's existing `.env` file and possibly make him lose some env values. Maybe we can add a prompt? (`You have an existing .env file. Are you sure you want to overwrite?`)

Reply:
I added this prompt when the file exists and it can be skipped with `--yes` or `-y`.